src/flinkSql/flinkLanguageClientManager.ts

```diff
@@ -0,0 +1,448 @@
+    }
+    return FlinkLanguageClientManager.instance;
+  }
+
+  private constructor() {
+    this.registerListeners();
+  }
+
+  private registerListeners(): void {
+    // Listen for user opening a Flink SQL file
+    this.disposables.push(
+      workspace.onDidOpenTextDocument(async (document) => {
+        if (document.languageId === "flinksql") {
+          logger.debug("Flink SQL file opened, checking for language client");
+          await this.maybeStartLanguageClient();
+        }
+      }),
+    );
+
+    // Listen for CCloud authentication
+    this.disposables.push(
+      ccloudConnected.event(async (connected) => {
+        if (connected) {
+          await this.maybeStartLanguageClient();
+        } else {
+          logger.debug("CCloud auth session invalid, stopping Flink language client");
+          this.cleanupLanguageClient();
+        }
+      }),
+    );
+
+    // Monitor Flink settings changes
+    this.disposables.push(
+      workspace.onDidChangeConfiguration(async (e) => {
+        if (e.affectsConfiguration("confluent.flink")) {
+          if (this.languageClient) {
+            await this.notifyConfigChanged();
+          } else {
+            await this.maybeStartLanguageClient();
+          }
+        }
+      }),
+    );
+
+    // Monitor the Flink enabled setting
+    this.disposables.push(
+      workspace.onDidChangeConfiguration(async (e) => {
+        if (e.affectsConfiguration(ENABLE_FLINK)) {
+          const isFlinkEnabled = getContextValue(ContextValues.flinkEnabled);
+          if (isFlinkEnabled) {
+            await this.maybeStartLanguageClient();
+          } else {
+            logger.debug("Flink is disabled in settings, stopping language client");
+            this.cleanupLanguageClient();
+          }
+        }
+      }),
+    );
+  }
+
+  /** Get the global/workspace settings for Flink, if any */
+  public getFlinkSqlSettings(): FlinkSqlSettings {
+    const config: WorkspaceConfiguration = workspace.getConfiguration();
+    const defaultPoolId: string = config.get(FLINK_CONFIG_COMPUTE_POOL, "");
+    const defaultDatabase: string = config.get(FLINK_CONFIG_DATABASE, "");
+    logger.debug("pool/db =>", defaultPoolId, defaultDatabase);
+    return {
+      database: defaultDatabase,
+      computePoolId: defaultPoolId,
+    };
+  }
+
+  /** Verify that Flink is enabled + the compute pool id setting exists and is in an environment we know about */
+  public async validateFlinkSettings(): Promise<boolean> {
+    const isFlinkEnabled = getContextValue(ContextValues.flinkEnabled);
+    if (!isFlinkEnabled) {
+      logger.debug("Flink is not enabled in settings, skipping configuration prompt");
+      return false;
+    }
+    const { computePoolId } = this.getFlinkSqlSettings();
+    if (!computePoolId) {
+      logger.debug("Flink compute pool not set");
+      await this.promptChooseDefaultComputePool();
+      return false;
+    }
+
+    logger.debug("Flink compute pool is:", computePoolId);
```

Comment:
Don't know if you want this level of debug logs everywhere -- seems useful for when really needed but maybe on the high noise side still